src/eko/runner/operators.py

```diff
@@ -16,7 +16,9 @@ def retrieve(
     elements = []
     for head in headers:
         inv = parts if isinstance(head, Evolution) else parts_matching
-        elements.append(inv[head])
+        op = inv[head]
+        assert isinstance(op, Operator)
+        elements.append(op)
```

Comment:
this is point 3 in [in my list](https://github.com/NNPDF/eko/pull/291#issuecomment-1645978114) (I hope you can read below the strike :upside_down_face: )

Reply:
Ok, you're right. I would have liked to make a generic constant for the `contentless`, but I thought it would have been too much for Python types (I should also derive the type from the const generic...).

Just to make slightly more explicit, check:
```suggestion
        op = inv[head]
        assert op is not None
        elements.append(op)
```
It will be sufficient to infer that is an `Operator`, and passing less information we'll rely more on the type checking.